pkg/imagestore/imagestore.go

```diff
@@ -154,8 +178,50 @@ func (s *rhcosStore) Populate(ctx context.Context) error {
 	return errs.Wait()
 }
 
-func (s *rhcosStore) PathForParams(imageType, version, arch string) string {
-	return filepath.Join(s.dataDir, fmt.Sprintf("rhcos-%s-%s-%s.iso", imageType, version, arch))
+func (s *rhcosStore) PathForParams(imageType, openshiftVersion, arch string) string {
+	var version string
+	for _, entry := range s.versions {
+		if entry["openshift_version"] == openshiftVersion && entry["cpu_architecture"] == arch {
```

Comment:
do we need fallback to default arch here in case it's missing? i.e. empty means x86. or is it already validated/handled?

Reply:
we plan to use fallback for day2, not sure if this api knows if it's for day1 or day2

Reply:
It does not, but we create the url in assisted-service so we should be able to do whatever defaulting we need on that side.